### ISSUE TITLE
Java and Kotlin code generation ignores `@include`

### ISSUE THREAD

**Issue by morganlutz** [apollographql/apollo-android#1883]
**Summary**
I am trying to upgrade from `1.0.0` to the latest version of apollo-android, `1.2.2`. However, the Java and Kotlin generated code seem to ignore the `@include` logic that I have defined in my fragment. The full fragment definition is included in the description.

The fragment optionally includes extra details using `@include(if: withDetails: Boolean)`. In the feed I don't need the extra details. In a detail page I do need the extra details. Some of the extra details are marked as `NON_NULL` in the schema, because if I request them then I should receive them.

Previously, the Java generated code marked these additional details as `@Nullable` regardless of whether `NON_NULL` was present in the schema. Now, it marks these objects as `@NotNull`.

I saw that apollo-android supports Kotlin now, so I tried migrating to Kotlin code generation to see if it was any different, but the extra details in the generated data class are non-null instead of optional. Whenever I make a call to fetch the object without the extra details, it throws an exception / fails to parse the http response.

I think this is a bug since it was working previously and the same fragment definition works well in the iOS app. If it's not a bug - What is the right way to define a fragment to optionally include attributes? Or, is there a recommended work-around?

**Description**
`@include` documentation: https://graphql.org/learn/queries/#directives
Schema - https://api.kivaws.org/graphql

Fragment definition:

```
fragment LoanFragment on LoanBasic {
    id
    status
    name
    use
    loanAmount
    plannedExpirationDate
    loanFundraisingInfo {
        fundedAmount
        reservedAmount
    }
    image {
        url(presetSize: loan_retina)
    }
    geocode {
        country {
            isoCode
            name
        }
    }
    ... on LoanBasic @include(if: $withDetails) {
        raisedDate
        description
        whySpecial
        activity {
            name
        }
        sector {
            id
            name
        }
        hasCurrencyExchangeLossLenders
        terms {
            lenderRepaymentTerm
        }
        repaymentInterval
        paidAmount
        userProperties {
            lentTo
        }
        isMatchable
        matcherName
    }
    ... on LoanPartner @include(if: $withDetails) {
        partnerName
        partner {
            id
            image {
                url(customSize: "h300")
            }
            loanAlertText
            chargesFeesInterest
            riskRating
        }
    }
}
```

A specific example using `Sector`, which we do not fetch in the feed, but do fetch on a detail page:

Version `1.0.0` generated Java code:

```
  /**
   * The sector is a more general classification of the loan than activity
   */
  @Nullable Sector sector();
```

Version `1.2.2` generated Java code:

```
  /**
   * The sector is a more general classification of the loan than activity
   */
  @NotNull Sector sector();
```

Version `1.2.2` generated Kotlin code:
```
  /**
   * The sector is a more general classification of the loan than activity
   */
  val sector: Sector,
```

Additionally, when I upgraded, the Java generated code failed to compile. I wrote a separate ticket for this: https://github.com/apollographql/apollo-android/issues/1884

**Version**
`1.2.2`

**Comment by sav007** [comment]
Thanks for reporting the issue. It looks we missing logic to read conditionally inline fragment.
For instance, having this query:
```
query TestQuery($withDetails: Boolean!) {
  hero {
    id
    ... on Human @include(if: $withDetails) {
        name
        homePlanet
    }
    ... on Droid @include(if: $withDetails) {
        name
        primaryFunction
    }
  }
}
```

We generate:
```
ResponseField.forInlineFragment("__typename", "__typename", listOf("Human", "Droid"))
```

But this is what we should generate instead:
```
ResponseField.forInlineFragment("__typename", "__typename", listOf("Human", "Droid"),  listOf(ResponseField.Condition.booleanCondition("withDetails", false)))
```

But just to be clear, for your example:
```
@NotNull Sector sector();
```
is correct as I assume it's defined as non-nullable. Having directives for inline fragments should only impact on how those fragments been read rather than making all of them nullable.

**Comment by morganlutz** [comment]
Hi @sav007. Thanks for the quick response.

>... is correct as I assume it's defined as non-nullable.

Yep - It is defined as non-nullable in the schema. But I don't request it all of the time... How does it work if `withDetails` is false and I don't receive `Sector`? For example, if I migrate to Kotlin and the generated data class looks like this?:

```
data class Loan(
    val id: Int,
    ....
    val sector: Sector
)
```

For now I refactored to use separate queries instead of `@includes` and we successfully upgraded. ⭐️
